### ISSUE TITLE
What behaviour should occur when all columns are hidden?

### ISSUE THREAD

**Issue by tdwright** [tdwright/contabs#19]
Relating to #13 - what behaviour do we want to happen when a data type has public properties (thereby not triggering the exception added in #18), but where the resulting columns have all been manually hidden?

**Comment by rahl0b10** [comment]
Before I commit to a position, I would like to know your reasoning behind having `Columns` and `_colsShown`. At first glance I'm inclined to believe it's a YAGNI issue. Unless it currently has a use-case I do not recognize I'd be inclined to simply not add hidden columns to the `Columns` property in the first place. 
```
public List<Column> Columns { get; set; }
```
~~internal List<Column> _colsShown => Columns.Where(c => !c.Hide).ToList();~~
```
private Table()
        {
            TableStyle = Style.Default;
            var props = typeof(T).GetTypeInfo().DeclaredProperties;
            Columns = props
                .Where(p => p.GetMethod.IsPublic)
                .Select(p => new Column(p.PropertyType, p.Name))
                .Where(c => !c.Hide)
                .ToList();
            
            if (!Columns.Any()) throw new PublicPropertiesNotFoundException();
        }
```

Since _colsShown isn't part of the API, it wouldn't create a breaking change.

**Comment by tdwright** [comment]
Hi @rahl0b10,

The idea behind this design is that we may want to generate a table for certain fields of a pre-existing type. In this case, we'd (indirectly) call the constructor on the type in it's entirety, before marking the `Hide` property of some columns as true.

For instance, we may want to do as follows:

```
var listOfData = new List<SomeExistingType>
{
	new SomeExistingType{String1 = "Hello", String2="Baboon", String3 = "World"}
};

var table = ConTabs.Table<SomeExistingType>.Create(listOfData);
table.Columns[1].Hide = true;

Console.WriteLine(table.ToString());
```

And because we've hidden a column, we get the following:

```
+---------+---------+
| String1 | String3 |
+---------+---------+
| Hello   | World   |
+---------+---------+
```
_Note that "Baboon" isn't printed._

All the examples and tests contained in the repo use types created _ad hoc_, solely as vehicles for the data destined for the table. In real usage, these tables are much more likely to receive `IEnumerable`s of types whose main purpose is elsewhere in the codebase. The flexibility to hide certain columns (or format them) is therefore quite important.

Hope this makes sense?

**Comment by rahl0b10** [comment]
Hi @tdwright ,

That does make sense, thank you!  In that case, I think I'd fall on the side of throwing a different exception.

**Comment by Chandler-Davidson** [comment]
I think that it should throw a unique exception that provides a meaningful suggestion.

**Comment by tdwright** [comment]
Fixed in #49
